### DISCUSSION TITLE
Polygon Polygon Intersection

### DISCUSSION THREAD

**Discussion by kurt-rhee** [rapidsai/cuspatial#1270] [Q&A] [answered]
Hello, 

I am looking for a library that can do many polygon-polygon intersections via the GPU, as far as I could tell cuspatial does not do polygon-polygon intersection, but I wanted to ask first just to be sure.

Thank you in advance!

**Comment by jarmak-nv** [comment, marked as answer]
Hi @kurt-rhee - thanks for checking out cuSpatial!

We can do the [binary intersects operation](https://docs.rapids.ai/api/cuspatial/stable/api_docs/geopandas_compatibility/#cuspatial.GeoSeries.intersects) between polygons, but if you're looking for the output geometry intersection between polygons then right now that's not supported - you're correct.

If your polygons don't have holes/you don't expect multipolygons you may be able to piece something together using the [LineString Intersection API](https://docs.rapids.ai/api/cuspatial/stable/user_guide/cuspatial_api_examples/#Linestring-Intersections), following something like:
- Extract the boundaries from your polygons and convert them to into LineStrings
- Perform intersection operation
- Combine the outputted LineStrings back into a polygon

**Comment by kurt-rhee** [comment]
Awesome thank you for the quick response!
